pandas/core/reshape/melt.py

```diff
@@ -0,0 +1,386 @@
+# pylint: disable=E1101,E1103
+# pylint: disable=W0703,W0622,W0613,W0201
+import numpy as np
+
+from pandas.core.dtypes.common import is_list_like
+from pandas import compat
+from pandas.core.categorical import Categorical
+
+from pandas.core.frame import DataFrame
+from pandas.core.index import MultiIndex
```

Comment:
you can do this now, or can in a followup. These top-level imports needs to be from the pandas namespace; some can be ``ABC``, e.g. ``ABCMultiIndex``, which you can import here. The others should be imported *inside* the functions themselves if needed. This allows this module to be imported irrespective of its import ordering.